src/class_sqlThread.py

```diff
@@ -15,128 +15,122 @@
     import paths
     import queues
     import state
+    from tr import _translate
     from addresses import encodeAddress
     from bmconfigparser import BMConfigParser
     from debug import logger
-    from tr import _translate
 except ImportError:
     from . import helper_sql, helper_startup, paths, queues, state
     from .addresses import encodeAddress
     from .bmconfigparser import BMConfigParser
     from .debug import logger
     from .tr import _translate
 
+# pylint: disable=attribute-defined-outside-init,protected-access
+root_path = os.path.dirname(os.path.dirname(__file__))
 
-class sqlThread(threading.Thread):
-    """A thread for all SQL operations"""
 
-    def __init__(self):
-        threading.Thread.__init__(self, name="SQL")
+def connection_build(file_name):
+    """
+        Stablish SQL connection
+    """
+    conn = sqlite3.connect(state.appdata + file_name)
+    conn.text_factory = str
+    cur = conn.cursor()
+    return conn, cur
 
-    def run(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-statements
-        """Process SQL queries from `.helper_sql.sqlSubmitQueue`"""
-        helper_sql.sql_available = True
-        self.conn = sqlite3.connect(state.appdata + 'messages.dat')
-        self.conn.text_factory = str
-        self.cur = self.conn.cursor()
 
-        self.cur.execute('PRAGMA secure_delete = true')
+class UpgradeDB(object):
+    """
+        Upgrade Db with respect to versions
+    """
+    conn, cur = connection_build('messages.dat')
 
-        # call create_function for encode address
-        self.create_function()
+    def __init__(self):
+        self._current_level = None
+        self.max_level = 11
+        # self.conn, self.cur = connection_build('messages.dat')
```

Comment:
make another class which overrides this to point to in-memory database, this can then be used for tests. You can call it something like `class MockDB(UpgradeDB)`

Reply:
Done

Reply:
Done